exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceConfiguration.java

```diff
@@ -60,6 +59,16 @@ public abstract class TraceConfiguration {
           .put("thread.id", "/tid")
           .build();
 
+  private Supplier<String> projectIdProvider;
```

Comment:
* An auto-value class should not have concrete fields or setters.
* This should be named `projectIdSupplier` to be consistent with the type name.

One way to do this would be
1. Remove the field
2. Add an abstract method `getProjectIdSupplier` to `TraceConfiguration`
3. Change the existing abstract method `getProjectId` to a concrete, [memoized](https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/pull/193/files?diff=split&w=1) method that calls `getProjectIdSupplier().get()`.
4. Add a concrete method `setProjectId` to `TraceConfiguration.Bulder` that calls `setProjectIdSupplier(() -> id)`.